pkgs/by-name/fe/fex/package.nix

```diff
@@ -11,19 +11,98 @@
+  runCommand,
+  writeText,
+  pkgsCross,
+  libclang,
+  libllvm,
+  alsa-lib,
+  libdrm,
+  libGL,
+  wayland,
+  xorg,
 }:
 
+let
+  pkgsCross32 = pkgsCross.gnu32;
+  pkgsCross64 = pkgsCross.gnu64;
+  devRootFSBase = buildEnv {
+    name = "fex-dev-rootfs";
+    paths = [
+      # later stdenv seems to have shuffled around the headers
+      # pkgsCross64.stdenv.cc.libc_dev
+      # pkgsCross32.stdenv.cc.libc_dev
+      # pkgsCross64.stdenv.cc.cc
+      # pkgsCross32.stdenv.cc.cc
+      pkgsCross64.gcc13Stdenv.cc.libc_dev
+      pkgsCross32.gcc13Stdenv.cc.libc_dev
+      pkgsCross64.gcc13Stdenv.cc.cc
+      pkgsCross32.gcc13Stdenv.cc.cc
+
+      alsa-lib.dev
+      libdrm.dev
+      libGL.dev
+      wayland.dev
+      xorg.libX11.dev
+      xorg.libxcb.dev
+      xorg.libXrandr.dev
+      xorg.libXrender.dev
+      xorg.xorgproto
+    ];
+    ignoreCollisions = true;
+    pathsToLink = [
+      "/include"
+      "/lib"
+    ];
+  };
+
+  devRootFS = runCommand "fex-dev-rootfs2" {} ''
+    mkdir -p $out/usr
+    ln -s ${devRootFSBase}/* $out/
+    ln -s ${devRootFSBase}/include $out/usr/
+  '';
+
+  toolchain32 = writeText "toolchain_nix_x86_32.txt" "
+    set(CMAKE_EXE_LINKER_FLAGS_INIT \"-fuse-ld=lld\")
+    set(CMAKE_MODULE_LINKER_FLAGS_INIT \"-fuse-ld=lld\")
+    set(CMAKE_SHARED_LINKER_FLAGS_INIT \"-fuse-ld=lld\")
+    set(CMAKE_SYSTEM_PROCESSOR i686)
+    set(CMAKE_C_COMPILER clang)
+    set(CMAKE_CXX_COMPILER clang++)
+    set(CLANG_FLAGS \"-nodefaultlibs -nostartfiles -lstdc++ -target i686-linux-gnu -msse2 -mfpmath=sse --sysroot=${devRootFS} -iwithsysroot/usr/include\")
+    set(CMAKE_C_COMPILER ${pkgsCross32.buildPackages.clang}/bin/i686-unknown-linux-gnu-clang)
+    set(CMAKE_CXX_COMPILER ${pkgsCross32.buildPackages.clang}/bin/i686-unknown-linux-gnu-clang++)
+    set(CMAKE_C_FLAGS \"\${CMAKE_C_FLAGS} \${CLANG_FLAGS}\")
+    set(CMAKE_CXX_FLAGS \"\${CMAKE_CXX_FLAGS} \${CLANG_FLAGS}\")
+  ";
+
+  toolchain = writeText "toolchain_nix_x86_64.txt" "
+    set(CMAKE_EXE_LINKER_FLAGS_INIT \"-fuse-ld=lld\")
+    set(CMAKE_MODULE_LINKER_FLAGS_INIT \"-fuse-ld=lld\")
+    set(CMAKE_SHARED_LINKER_FLAGS_INIT \"-fuse-ld=lld\")
+    set(CMAKE_SYSTEM_PROCESSOR x86_64)
+    set(CMAKE_C_COMPILER clang)
+    set(CMAKE_CXX_COMPILER clang++)
+    set(CLANG_FLAGS \"-nodefaultlibs -nostartfiles -lstdc++ -target x86_64-linux-gnu --sysroot=${devRootFS} -iwithsysroot/usr/include\")
+    set(CMAKE_C_COMPILER ${pkgsCross64.buildPackages.clang}/bin/x86_64-unknown-linux-gnu-clang)
+    set(CMAKE_CXX_COMPILER ${pkgsCross64.buildPackages.clang}/bin/x86_64-unknown-linux-gnu-clang++)
+    set(CMAKE_C_FLAGS \"\${CMAKE_C_FLAGS} \${CLANG_FLAGS}\")
+    set(CMAKE_CXX_FLAGS \"\${CMAKE_CXX_FLAGS} \${CLANG_FLAGS}\")
+  ";
+in
 llvmPackages.stdenv.mkDerivation (finalAttrs: {
   pname = "fex";
-  version = "2505";
+  # version = "2505";
+  version = "5811914a784cc6fdfdc90242ab7f8ea6ff39ec38";
 
   src = fetchFromGitHub {
-    owner = "FEX-Emu";
+    owner = "neobrain";
```

Comment:
This is currently required for https://github.com/neobrain/FEX/commit/5811914a784cc6fdfdc90242ab7f8ea6ff39ec38 , which will be included in the next FEX release (2506).